---
title: "Final Project"
author: "Leonid Sidorov, Vasiliy Zubarev, Yawar Habib"
output:
  pdf_document:
    latex_engine: pdflatex
    toc: yes
    number_sections: yes
    toc_depth: 3
    fig_width: 5
    fig_height: 4
geometry: margin = 2cm
lof: no
fontsize: 10pt
documentclass: article
classoption: a4paper
---

```{r setup, include=FALSE}
# To erase all graphs
graphics.off()
# To erase objects from the working space - Clean up of the memory
rm(list = ls())
# use of the constraint 'set-to-zero' for ANOVAs
options(contrasts=c('contr.treatment','contr.poly'))

suppressPackageStartupMessages(
  {
    library(qtl)
    library(lattice)
    library(tidyverse)
  })
#can also use 'contr.sum' for a 'sum-to-zero' constraint

# postscript figures with colours
trellis.device(device = postscript,color = T)

knitr::opts_chunk$set(echo = TRUE)
```


# Data

Firstly, we import the data.

```{r data_import, echo=TRUE}
QTL_LR4RalstoDataRaw <- read.cross("csvs", genfile="data/LR4_CGPM_avril 2007_geno.csvs", 
                                     phefile="data/LR4_Ralsto_AllScores.csvs", sep=";", dec=",")

#genfile: file containing genotypes, phefile:file containing phenotypes.

QTL_LR4RalstoDataRaw <- convert2riself(QTL_LR4RalstoDataRaw)

#The function convert2riself converts the type of crossing indicating that the analysis was made on RILs obtained by self-fertilization.
summary(QTL_LR4RalstoDataRaw)
```

# ANALYSIS OF THE QUALITY OF PHENOTYPIC AND GENOTYPIC DATA

<!--I took the text from the script QTL_LR5Ralsto_Complete.R, so we will have to change it-->

Graph 1: Evaluation of the quality of genotypic data: visualization of missing data.
Graph 2: Visualization of the genetic map
Graph 3 to 15 (phe 1 to phe 13): Distribution of phenotypic data for the different traits evaluated within the population of RILs

```{r graphic_visualistion, echo=FALSE, fig.align='center', fig.cap=''}
plot(QTL_LR4RalstoDataRaw)
```

Detail of the genetic map.

```{r detail_gen_map, echo=FALSE}
plot.map(QTL_LR4RalstoDataRaw, show.marker.names=TRUE)
```

Details of missing genotypic data.

```{r missing_genotypic_data, echo=FALSE}
plotMissing(QTL_LR4RalstoDataRaw, reorder=TRUE)
```

## Further analysis of the quality of genotypes

* Calculation of recombination fractions.

```{r rf, echo=FALSE}
QTL_LR4RalstoDataRaw <- est.rf(QTL_LR4RalstoDataRaw)
plotRF(QTL_LR4RalstoDataRaw)
```

The plot above informs us that there is no problem with positioning markers on chromosomes.

* Calculation of probabilities of genotyping errors

```{r}
QTL_LR4RalstoDataRaw <- calc.errorlod(QTL_LR4RalstoDataRaw, error.prob=0.1)
top.errorlod(QTL_LR4RalstoDataRaw)

plotGeno(QTL_LR4RalstoDataRaw)

#Observed marker genotype data for RIL population. White and black circles correspond to the genotypes AA, and BB (AB are ignored due to convert2riself), respectively. 
#Genotypes flagged by red squares would be indicated to be likely genotyping errors. Here there is none.
#Blue crosses indicate recombination events.
```

## More detailed analysis of phenotypes

In advance to QTL analysis, we performed ANOVA in order to see if there is an effect of the "line" factor on the phenotypic trait analyzed, and to computed adjusted means corrected from the block or environmental effect.

```{r echo=TRUE}
Pheno_Score5dpi<-data.frame(indice=QTL_LR4RalstoDataRaw$pheno$scoreajst5dpi)
str(Pheno_Score5dpi)
shapiro.test(Pheno_Score5dpi$indice)

Pheno_Score14dpi<-data.frame(indice=QTL_LR4RalstoDataRaw$pheno$scoreajst14dpi)
str(Pheno_Score14dpi)
shapiro.test(Pheno_Score14dpi$indice)

# I do not really understand where to take Moyenne.logcfy column

# Pheno_cfu<-data.frame(indice=QTL_LR5RalstoDataRaw$pheno$Moyenne.logcfu)
# Pheno_cfu
# str(Pheno_cfu)
# shapiro.test(Pheno_cfu$indice)
```

For 5 dpi we do not have normal data.

# DETECTION DE QTL PAR SCANONE: mr method ie Marker regression or ANOVA at each marker (Single QTL genome scan)

As it was stated in the task, we would be able to detect QTL from mean disease scores 5 and 14 dpi.

## Analysis on the disease score at 5dpi

```{r scanone_rm_5, echo=TRUE}
QTLT5_LR4Ralsto.mr <- scanone(QTL_LR4RalstoDataRaw, pheno.col=3, method="mr")
summary(QTLT5_LR4Ralsto.mr)
```

## Analysis on the disease score at 14dpi

```{r scanone_rm_14}
QTLT14_LR4Ralsto.mr<-scanone(QTL_LR4RalstoDataRaw, pheno.col=7,method="mr") #cfu
summary(QTLT14_LR4Ralsto.mr)
```

Marker MTE32 shows quite interesting LOD score - so probably it is linked to the QTL.


# Simple interval mapping by maximum likelihood (Single QTL genome scan)

Imputing the missing values.

```{r calc_miss_values}
QTL_LR4RalstoData<- calc.genoprob(QTL_LR4RalstoDataRaw, step=1, error.prob=0.01)
```

## Analysis on the disease score at 5dpi

* QTL detection

```{r qtl_detection_em}
QTLT5_LR4Ralsto.em<-scanone(QTL_LR4RalstoData, pheno.col=3,method="em") #T5
summary(QTLT5_LR4Ralsto.em) # only the lod max on each chromosome
```

* Graphic visualization

```{r plot_em}
plot(QTLT5_LR4Ralsto.em)
```

* Graphic visualization of the results of the genotype effect at the marker closest to the major QTL peak

```{r gen_5_graph_vis}
max(QTLT5_LR4Ralsto.em)
find.marker(QTL_LR4RalstoData, chr="LG5", pos=15.1) #Goal: find the marker closest to the QTL's lod peak on ch5

effectplot(QTL_LR4RalstoData, mname1="LG5@15.1", pheno.col=3, main="LR5 Ralstonia - QTL LG5 effect")

plotPXG(QTL_LR4RalstoData, marker="MTE32", pheno.col=3, main="LR5 Ralstonia - QTL LG5 effect")
```

* Calculation of QTL confidence intervals (CI) and identification of flanking markers

```{r}
lodint(QTLT5_LR4Ralsto.em, chr="LG5", expandtomarkers=TRUE)
lodint(QTLT5_LR4Ralsto.em, chr="LG5", expandtomarkers=TRUE, drop=1) #20cM between MTE32 and MTE34
lodint(QTLT5_LR4Ralsto.em, chr="LG5", drop=1) #20cM between MTE32 and MTE34
# the CI is given in positions, not in markers:
ICfinT5<-lodint(QTLT5_LR4Ralsto.em, chr="LG5", drop=1)#12cM
```

* Graphic visualization listing all the information concerning the major QTL

```{r graph_vis_all_5_em}
plot(QTLT5_LR4Ralsto.em,chr="LG5",
     show.marker.names=T,
	xlab="Map Position on LG5 (cM)",
     ylab="LOD Score LR4Ralsto_ScoreT5")
abline(h=3,col="darkgreen",lty=2)
abline(v=c(ICfinT5$pos), col=c("blue", "red", "blue"), lty=c(2,1,2))
```

## Analysis on the disease score at 14dpi

* QTL detection

```{r qtl_detection_em}
QTLT14_LR4Ralsto.em <- scanone(QTL_LR4RalstoData, pheno.col=7,method="em") #T14
summary(QTLT14_LR4Ralsto.em) # only the lod max on each chromosome
```

* Graphic visualization

```{r plot_em}
plot(QTLT14_LR4Ralsto.em)
```

* Graphic visualization of the results of the genotype effect at the marker closest to the major QTL peak

```{r gen_5_graph_vis, echo=TRUE, message=FALSE, warning=FALSE}
max(QTLT14_LR4Ralsto.em)
find.marker(QTL_LR4RalstoData, chr="LG5", pos=17) #Goal: find the marker closest to the QTL's lod peak on ch5

effectplot(QTL_LR4RalstoData, mname1="LG5@17", pheno.col=7, main="LR5 Ralstonia - QTL LG5 effect")

plotPXG(QTL_LR4RalstoData, marker="MTE32", pheno.col=7, main="LR5 Ralstonia - QTL LG5 effect")
```

* Calculation of QTL confidence intervals (CI) and identification of flanking markers

```{r echo=TRUE}
lodint(QTLT14_LR4Ralsto.em, chr="LG5", expandtomarkers=TRUE)
lodint(QTLT14_LR4Ralsto.em, chr="LG5", expandtomarkers=TRUE, drop=1) #20cM between MTE32 and MTE34
lodint(QTLT14_LR4Ralsto.em, chr="LG5", drop=1) #20cM between MTE32 and MTE34
# the CI is given in positions, not in markers:
ICfinT14<-lodint(QTLT14_LR4Ralsto.em, chr="LG5", drop=1)#12cM
```

* Graphic visualization listing all the information concerning the major QTL

```{r graph_vis_all_14_em}
plot(QTLT14_LR4Ralsto.em,chr="LG5",
     show.marker.names=T,
	xlab="Map Position on LG5 (cM)",
     ylab="LOD Score LR4Ralsto_ScoreT14")
abline(h=3,col="darkgreen",lty=2)
abline(v=c(ICfinT14$pos), col=c("blue", "red", "blue"), lty=c(2,1,2))
```

## Major QTL IC comparisons on LG5 for 5 dpi and at 14 dpi

```{r QTL_IC_comparisons}
plot(QTLT5_LR4Ralsto.em, QTLT14_LR4Ralsto.em, chr="LG5",
     show.marker.names=T,
	xlab="Map Position on LG5 (cM)",
     ylab="LOD Score LR4Ralsto")
abline(h=3,col="darkgreen",lty=2)
abline(v=c(ICfinT5$pos), col=c("black"), lty=c(2,1,2))
abline(v=c(ICfinT14$pos), col=c("red"), lty=c(2,1,2))
```

# Imputation method or anova after imputation of missing genotypes (Single QTL genome scan)

* Imputation of the missing data

```{r miss_value_imp}
QTL_LR4RalstoDataImp<- sim.geno(QTL_LR4RalstoData, step=2, n.draws=16, error.prob=0.01)    
```

## QTL detection for 5 dpi

```{r qtl_detection_imp}
QTLT5_LR4Ralsto.imp<-scanone(QTL_LR4RalstoDataImp, pheno.col=3,method="imp")#T5
QTLT5_LR4Ralsto.imp
summary(QTLT5_LR4Ralsto.imp)
```

* Graphic visualization comparing the major QTL on LG5 detected by the imp and em method.

```{r compar_of_em_and_imp}
plot(QTLT5_LR4Ralsto.imp, QTLT5_LR4Ralsto.em, chr="LG5", col=c("green", "pink"), ylab="LodScore Ralsto LR4 T5")
```

## QTL detection for 14 dpi

```{r qtl_detection_imp}
QTLT14_LR4Ralsto.imp<-scanone(QTL_LR4RalstoDataImp, pheno.col=7,method="imp")#T14
QTLT14_LR4Ralsto.imp
summary(QTLT14_LR4Ralsto.imp)
```

* Graphic visualization comparing the major QTL on LG5 detected by the imp and em method.

```{r compar_of_em_and_imp}
plot(QTLT14_LR4Ralsto.imp, QTLT14_LR4Ralsto.em, chr="LG5", col=c("green", "pink"), ylab="LodScore Ralsto LR4 T14")
```

# DETECTION OF QTL with SCANONE (Single QTL genome scan): hk method Haley-Knott regression

## Detection of QTL with the Haley-Knott regression method

```{r}
QTLall_LR4Ralsto.hk <- scanone(QTL_LR4RalstoData, pheno.col=2:10,method="hk")      # all cfu + scoredpi phenotypes
summary(QTLall_LR4Ralsto.hk)
```

* Graphic visualization

```{r}
plot(QTLall_LR4Ralsto.hk, lodcolumn=1:9, col=c("blue", "red", "green"))
plot(QTLall_LR4Ralsto.hk, lodcolumn=1:3, col=c("blue", "red", "green"))
plot(QTLall_LR4Ralsto.hk, lodcolumn=2:4, col=c("blue", "red", "green"))
plot(QTLall_LR4Ralsto.hk, lodcolumn=4:6, col=c("pink", "orange", "grey"))
plot(QTLall_LR4Ralsto.hk, lodcolumn=7:9, col=c("violet", "cyan", "yellow"))
```

```{r}
# why do we need so many summaries?

# summary(QTLall_LR4Ralsto.hk,format="onepheno",lodcolumn=1)  # cfu
# summary(QTLall_LR4Ralsto.hk,format="onepheno",lodcolumn=2)  # T3
# summary(QTLall_LR4Ralsto.hk,format="onepheno",lodcolumn=3)  # T5
# summary(QTLall_LR4Ralsto.hk,format="onepheno",lodcolumn=5)  # T7
# summary(QTLall_LR4Ralsto.hk,format="onepheno",lodcolumn=7)  # T14
```

## Calculation of lod thresholds by permutation

```{r}
# QTL_LR4RalstoData.operm.hk<-scanone(QTL_LR4RalstoData, pheno.col=2:10, method="hk", n.perm=1000)
# save(QTL_LR4RalstoData.operm.hk, file = "QTL_LR4RalstoData.operm.hk.RData")

load("QTL_LR4RalstoData.operm.hk.RData")

summary(QTL_LR4RalstoData.operm.hk, alpha=0.05)
```

```{r}
summary(QTL_LR4RalstoData.operm.hk, alpha=0.1)
summary(QTL_LR4RalstoData.operm.hk, alpha=0.01)
```
```{r}
summary(QTLall_LR4Ralsto.hk, perms=QTL_LR4RalstoData.operm.hk, lodcolumn=1, alpha=0.05, pvalues=TRUE)
summary(QTLall_LR4Ralsto.hk, perms=QTL_LR4RalstoData.operm.hk, lodcolumn=2, alpha=0.05, pvalues=TRUE)
summary(QTLall_LR4Ralsto.hk, perms=QTL_LR4RalstoData.operm.hk, lodcolumn=3, alpha=0.05, pvalues=TRUE)
summary(QTLall_LR4Ralsto.hk, perms=QTL_LR4RalstoData.operm.hk, lodcolumn=4, alpha=0.05, pvalues=TRUE)
summary(QTLall_LR4Ralsto.hk, perms=QTL_LR4RalstoData.operm.hk, lodcolumn=5, alpha=0.05, pvalues=TRUE)
summary(QTLall_LR4Ralsto.hk, perms=QTL_LR4RalstoData.operm.hk, lodcolumn=6, alpha=0.05, pvalues=TRUE)
summary(QTLall_LR4Ralsto.hk, perms=QTL_LR4RalstoData.operm.hk, lodcolumn=7, alpha=0.05, pvalues=TRUE)
summary(QTLall_LR4Ralsto.hk, perms=QTL_LR4RalstoData.operm.hk, lodcolumn=8, alpha=0.05, pvalues=TRUE)
summary(QTLall_LR4Ralsto.hk, perms=QTL_LR4RalstoData.operm.hk, lodcolumn=9, alpha=0.05, pvalues=TRUE)
```
We will continue to work with 5 and 14 dpi. <!--But I am not actually sure if we have to use these values. Your thoughts?--->

## Calculation of QTL confidence intervals and identification of flanking markers

```{r}
lodint(QTLall_LR4Ralsto.hk, lodcolumn=2, chr="LG5", expandtomarkers=TRUE, drop=1)
lodint(QTLall_LR4Ralsto.hk, lodcolumn=2, chr="LG5", drop=1)

lodint(QTLall_LR4Ralsto.hk, lodcolumn=6, chr="LG5", expandtomarkers=TRUE, drop=1)
lodint(QTLall_LR4Ralsto.hk, lodcolumn=6, chr="LG5", drop=1)
```


# DETECTION OF QTL with covariates, Haley & Knott regression

## Analysis on the disease score at 5dpi.

```{r}
max(QTLall_LR4Ralsto.hk, lodcolumn=2)#T5
QTLT5_LR4Ralsto.hk<-scanone(QTL_LR4RalstoData, pheno.col=3, method="hk")
plot(QTLT5_LR4Ralsto.hk)
```

```{r}
max(QTLT5_LR4Ralsto.hk)
find.marker(QTL_LR4RalstoData, chr="LG5", pos=16)
```

* Using MTE32 as a cofactor

```{r}
g<-pull.geno(QTL_LR4RalstoData) [, "MTE32"]
mean(is.na(g))
```
About 3.7% of the individuals are not genotyped on this marker.

```{r}
gnum<-cbind(as.numeric(g==1), as.numeric(g==2))
QTLT5_LR4Ralsto.hk.MTE32 <- scanone(QTL_LR4RalstoData, pheno.col=3, method="hk", addcovar=gnum)
summary(QTLT5_LR4Ralsto.hk.MTE32)
```

```{r}
plot(QTLT5_LR4Ralsto.hk, QTLT5_LR4Ralsto.hk.MTE32, col=c("blue", "red"))
```

```{r}
lodint(QTLT5_LR4Ralsto.hk, chr="LG5", expandtomarkers=TRUE, drop=1)
```

### Do we have problems here?

```{r}
# QTLT5_LR4Ralsto.hk.MTE32operm <- scanone(QTL_LR4RalstoData, pheno.col=3, method="hk", addcovar=gnum, n.perm=1000)
# save(QTLT5_LR4Ralsto.hk.MTE32operm, file = "QTLT5_LR4Ralsto.hk.MTE32operm.RData")

load("QTLT5_LR4Ralsto.hk.MTE32operm.RData")


summary(QTLT5_LR4Ralsto.hk.MTE32operm)
```

### Using MTE33 as a cofactor

```{r}
g4<-pull.geno(QTL_LR4RalstoData) [, "MTE33"]
mean(is.na(g4))
```
About 18% of the individuals are not genotyped on this marker.

```{r}
QTLT5_LR4Ralsto.hk.MTE33<-scanone(QTL_LR4RalstoData, pheno.col=3, method="hk", addcovar=g4)
summary(QTLT5_LR4Ralsto.hk.MTE33)
```

```{r}
plot(QTLT5_LR4Ralsto.hk, QTLT5_LR4Ralsto.hk.MTE33, col=c("blue", "red"))
plot(QTLT5_LR4Ralsto.hk, QTLT5_LR4Ralsto.hk.MTE33, chr="LG5",col=c("blue", "red"))
plot(QTLT5_LR4Ralsto.hk, QTLT5_LR4Ralsto.hk.MTE32, QTLT5_LR4Ralsto.hk.MTE33, chr="LG5",col=c("blue", "darkgreen", "red"))
```

```{r}
lodint(QTLT5_LR4Ralsto.hk, chr="LG5", drop=1)
lodint(QTLT5_LR4Ralsto.hk.MTE32, chr="LG5", drop=1)
lodint(QTLT5_LR4Ralsto.hk.MTE33, chr="LG5", drop=1)
```

### Graphic representation of QTL on genetic map

```{r}
qtlGraphMap<-makeqtl(QTL_LR4RalstoData, chr=c("LG5"), pos=16, what="prob")

plot(qtlGraphMap, main="LR5Ralsto T5")
effectplot(QTL_LR4RalstoData, mname1="LG5@16", pheno.col=3, main="LR5 ralsto T5 QTL LG5 effect")
plotPXG(QTL_LR4RalstoData, marker="MTE32", pheno.col=3, main="LR5 ralsto T5 QTL LG5 effect")
```


## Analysis on the disease score at 14dpi.

```{r}
max(QTLall_LR4Ralsto.hk, lodcolumn=6)#T5
QTLT14_LR4Ralsto.hk<-scanone(QTL_LR4RalstoData, pheno.col=7, method="hk")
plot(QTLT5_LR4Ralsto.hk)
```

```{r}
max(QTLT14_LR4Ralsto.hk)
find.marker(QTL_LR4RalstoData, chr="LG5", pos=18)
```
Here we have the same situation as for 5 dpi.


* Using MTE32 as a cofactor

```{r}
g<-pull.geno(QTL_LR4RalstoData) [, "MTE32"]
mean(is.na(g))
```
About 3.7% of the individuals are not genotyped on this marker.

```{r}
gnum<-cbind(as.numeric(g==1), as.numeric(g==2))
QTLT14_LR4Ralsto.hk.MTE32 <- scanone(QTL_LR4RalstoData, pheno.col=7, method="hk", addcovar=gnum)
summary(QTLT14_LR4Ralsto.hk.MTE32)
```

No marker is significant here as well.

```{r}
plot(QTLT14_LR4Ralsto.hk, QTLT14_LR4Ralsto.hk.MTE32, col=c("blue", "red"))
```

```{r}
lodint(QTLT14_LR4Ralsto.hk, chr="LG5", expandtomarkers=TRUE, drop=1)
```

### Do we have problems here?

```{r}
# QTLT14_LR4Ralsto.hk.MTE32operm <- scanone(QTL_LR4RalstoData, pheno.col=7, method="hk", addcovar=gnum, n.perm=1000)
# save(QTLT14_LR4Ralsto.hk.MTE32operm, file = "QTLT14_LR4Ralsto.hk.MTE32operm.RData")

load("QTLT14_LR4Ralsto.hk.MTE32operm.RData")


summary(QTLT5_LR4Ralsto.hk.MTE32operm)
```

### Using MTE33 as a cofactor

```{r}
g4<-pull.geno(QTL_LR4RalstoData) [, "MTE33"]
mean(is.na(g4))
```
As we stated earlier, ~18% of the individuals are not genotyped on this marker.

```{r}
QTLT14_LR4Ralsto.hk.MTE33<-scanone(QTL_LR4RalstoData, pheno.col=7, method="hk", addcovar=g4)
summary(QTLT14_LR4Ralsto.hk.MTE33)
```

```{r}
plot(QTLT14_LR4Ralsto.hk, QTLT14_LR4Ralsto.hk.MTE33, col=c("blue", "red"))
plot(QTLT14_LR4Ralsto.hk, QTLT14_LR4Ralsto.hk.MTE33, chr="LG5",col=c("blue", "red"))
plot(QTLT14_LR4Ralsto.hk, QTLT14_LR4Ralsto.hk.MTE32, QTLT14_LR4Ralsto.hk.MTE33, chr="LG5",col=c("blue", "darkgreen", "red"))
```

```{r}
lodint(QTLT14_LR4Ralsto.hk, chr="LG5", drop=1)
lodint(QTLT14_LR4Ralsto.hk.MTE32, chr="LG5", drop=1)
lodint(QTLT14_LR4Ralsto.hk.MTE33, chr="LG5", drop=1)
```

### Graphic representation of QTL on genetic map

```{r}
qtlGraphMap<-makeqtl(QTL_LR4RalstoData, chr=c("LG5"), pos=18, what="prob")

plot(qtlGraphMap, main="LR5Ralsto T5")
effectplot(QTL_LR4RalstoData, mname1="LG5@18", pheno.col=7, main="LR5 ralsto T5 QTL LG5 effect")
plotPXG(QTL_LR4RalstoData, marker="MTE32", pheno.col=7, main="LR5 ralsto T5 QTL LG5 effect")
```


It is interesting that we see more effect of this QTL. <!--Am I right, guys?-->

# DETECTION OF QTL with Multiple QTL Mapping

```{r}
geno.image(QTL_LR4RalstoDataRaw)
```
Imput missing values and compare results.
```{r}
m.imp <- mqmaugment(QTL_LR4RalstoData, minprob=1.0)
geno.image(m.imp)
```

Fit the Multiple QTL Mapping model without cofactors for 5dpi.
```{r}
abrev <- 'Disease index T5'        
phenotype <- 3
colonne <- phenotype-1

Phenotype.MQMnocofactor <- mqmscan(m.imp, pheno.col=phenotype, window.size=10, step.size=1)

```

```{r}
summary(Phenotype.MQMnocofactor, threshold=1.5)
```

```{r}
summary(QTLall_LR4Ralsto.hk,threshold=1.5, lodcolumn=colonne)
```
As we can see, LOD scores for LG4:loc72 and LG5.loc17 are similar between MQM and HK
```{r}
#subset only 5dpi from hk dataframe
QTLall_LR4Ralsto_5dpi.hk <- QTLall_LR4Ralsto.hk[c("chr","pos","scoreajst5dpi")]

plot(QTLall_LR4Ralsto_5dpi.hk,chr="LG5",   show.marker.names=T, col=c('blue'), xlab="Map Position on LG5 (cM)", ylab="LOD Score")

plot(Phenotype.MQMnocofactor, 
         chr='LG5', col=c('red'),add=T)

abline(h=2.5,lty=2)

legend('topright',col=c('blue','red'),lwd=2,legend=c(paste(abrev,'; ehk'),paste(abrev,'; mqm')))

```

##MQM model with cofactors
"Forward" cofactor search
```{r}
find.marker(m.imp, chr= summary(Phenotype.MQMnocofactor)$chr, pos= summary(Phenotype.MQMnocofactor)$pos)
index1 <- find.markerindex(m.imp, find.marker(m.imp, chr= summary(Phenotype.MQMnocofactor)$chr, pos= summary(Phenotype.MQMnocofactor)$pos))
index1
```

```{r}
set1 <- mqmsetcofactors(m.imp,cofactors=index1)
set1
``` 

Detect QTL by MQM with cofactors
```{r}
Phenotype.MQMset1 <- mqmscan(m.imp , set1, cofactor.significance=0.99, pheno.col=phenotype, window.size=10,step.size=1)
summary(Phenotype.MQMset1)
plot(mqmgetmodel(Phenotype.MQMset1))
```

```{r}
plot(QTLall_LR4Ralsto_5dpi.hk, chr="LG5",
     show.marker.names=T, col=c('red'),
     xlab="Map Position on LG5 (cM)",
     ylab="LOD Score"
     )
plot(Phenotype.MQMnocofactor, chr="LG5",
     col=c('blue'),
#	xlab="Map Position on LG5 (cM)",
#     ylab="LOD Score",
     add=T
     )
plot(Phenotype.MQMset1, chr="LG5",
     col=c('darkgreen'),
#	xlab="Map Position on LG5 (cM)",
#     ylab="LOD Score",
     add=T
     )
abline(h=3,lty=2)
legend('topright',col=c('blue','red','darkgreen'),lwd=2,legend=c(paste(abrev,'; mqm-0'),paste(abrev,'; ehk'),paste(abrev,'; mqm-1')))

```



